File: Data Structures & Algorithms/jump-game/submission-1.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        res = [False] * len(nums)
        res[0] = True

        for i in range(len(res)):
            if res[i] is True:
                
                for j in range(1,nums[i]+1):
                    if i+j >= len(res):
                        break
                    res[i+j] = True       
        return res[-1]

File: Data Structures & Algorithms/k-closest-points-to-origin/submission-1.java
class Solution {
    public int[][] kClosest(int[][] points, int k) {
        int[][] res = new int[k][2];
        int resSize = 0;

        if (k >= points.length) {
            return points;
        }

        PriorityQueue<int[]> pq =
            new PriorityQueue<>((p1, p2)
                                    -> Double.compare(Math.sqrt((p1[0] * p1[0]) + (p1[1] * p1[1])),
                                        Math.sqrt((p2[0] * p2[0]) + (p2[1] * p2[1]))));

        for (int[] p : points) {
            pq.add(p);
        }

        while (k > 0) {
            res[resSize] = pq.poll();
            resSize++;
            k--;
        }

        return res;
    }
}
